plugin/sampling/strategystore/static/strategy_store_test.go

```diff
@@ -465,18 +467,44 @@ func TestAutoUpdateStrategyErrors(t *testing.T) {
 }
 
 func TestServiceNoPerOperationStrategies(t *testing.T) {
+	// given setup of strategy store with no specific per operation sampling strategies
 	store, err := NewStrategyStore(Options{StrategiesFile: "fixtures/service_no_per_operation.json"}, zap.NewNop())
 	require.NoError(t, err)
 
-	s, err := store.GetSamplingStrategy(context.Background(), "ServiceA")
+	// expected response for ServiceA (which has probablistic service level sampling strategy)
+	// shall contain /health operation with default service level sampling strategy as well
+	expectedServiceAResponse, errServiceA := makeServiceStrategyResponseFromFile("fixtures/TestServiceNoPerOperationStrategies_response_for_ServiceA.json")
+	require.NoError(t, errServiceA)
+
+	strategy, err := store.GetSamplingStrategy(context.Background(), "ServiceA")
 	require.NoError(t, err)
-	assert.Equal(t, 1.0, s.OperationSampling.DefaultSamplingProbability)
+	require.NotNil(t, strategy.OperationSampling)
+
+	assert.Equal(t, expectedServiceAResponse, strategy)
 
-	s, err = store.GetSamplingStrategy(context.Background(), "ServiceB")
+	// expected response for ServiceB (which has ratelimiting service level sampling strategy)
+	// shall contain /health operation with sampling probability taken from default operation level strategy
+	expectedServiceBResponse, errServiceB := makeServiceStrategyResponseFromFile("fixtures/TestServiceNoPerOperationStrategies_response_for_ServiceB.json")
+	require.NoError(t, errServiceB)
+
+	strategy, err = store.GetSamplingStrategy(context.Background(), "ServiceB")
 	require.NoError(t, err)
+	require.NotNil(t, strategy.OperationSampling)
```

Comment:
since you're comparing with a snapshot, do you need this assert?

Reply:
Nop :)